### PR TITLE
pappl: update; modernize; format; move to by-name

### PR DESCRIPTION
**Pull request by NotAShelf**
- **pappl: 1.4.6 -> 1.4.7**
- **pappl: modernize derivation; add changelog**
- **pappl: format via nixfmt**
- **pappl: migrate to by-name**

Adds myself to maintainers, and performs maintenance on the pappl package
that was missing a maintainer following jonringer's... departure. Updated
the package to the 14 minute old update, formatted and moved to by-name.

If anyone can test Darwin support, I can try adding Darwin to the supported
platforms!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pa/pappl/package.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pappl";
+  version = "1.4.7";
+
+  src = fetchFromGitHub {
+    owner = "michaelrsweet";
+    repo = "pappl";
+    rev = "refs/tagsv${version}";
```
Comment (pbsds) [inline pkgs/by-name/pa/pappl/package.nix:24]:
```suggestion
    rev = "refs/tags/v${version}";
```

likely the cause of the ofborg 404 error

Comment (NotAShelf) [reply pkgs/by-name/pa/pappl/package.nix:24]: Whoops.

Comment (NotAShelf) [reply pkgs/by-name/pa/pappl/package.nix]: Fixed, thanks for catching that.

Though I had to squash it into the by-name commit, not very fond of that but it's the only way of *not* redoing the entire PR I think.

Comment (pbsds) [reply pkgs/by-name/pa/pappl/package.nix]: in the future it's easier if you start with the reformat commit

**Pull request merged**